src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -401,25 +425,66 @@ async def package_python_dist(
         ),
     )
 
+    # Find the source roots for the build-time 1stparty deps (e.g., deps of setup.py).
+    source_roots_result = await Get(
+        SourceRootsResult,
+        SourceRootsRequest(
+            files=[], dirs={PurePath(tgt.address.spec_path) for tgt in transitive_targets.closure}
+        ),
+    )
+    source_roots = tuple(sorted({sr.path for sr in source_roots_result.path_to_root.values()}))
+
+    # Get any extra build-time environment (e.g., native extension requirements).
+    build_env_requests = []
+    build_env_request_types = union_membership[DistBuildEnvironmentRequest]
+    for build_env_request_type in build_env_request_types:
```

Comment:
Note that, hypothetically at least, we support multiple build environments. E.g., a single setup.py could use mypyc *and* (in the future) cython...